miio/integrations/ihcooker/ihcooker.py

```diff
@@ -0,0 +1,565 @@
+            return bool(self.data["set_wifi_led"] == "01")
+        else:
+            return None
+
+    @property
+    def hardware_version(self) -> int:
+        """Hardware version."""
+        return int(self.data["version"][0:4], 16)
+
+    @property
+    def firmware_version(self) -> int:
+        """Firmware version."""
+        return int(self.data["version"][4:8], 16)
+
+
+class IHCooker(Device):
+    """Main class representing the induction cooker.
+
+    Custom recipes can be build with the profile_v1/v2 structure.
+    """
+
+    _supported_models = SUPPORTED_MODELS
+
+    @command(
+        default_output=format_output(
+            "",
+            "Mode: {result.mode}\n"
+            "Recipe ID: {result.recipe_id}\n"
+            "Recipe Name: {result.recipe_name}\n"
+            "Stage: {result.stage}\n"
+            "Stage Mode: {result.stage_mode}\n"
+            "Target Temp: {result.target_temp}\n"
+            "Temperature: {result.temperature}\n"
+            "Temperature upperbound: {result.temperature_upperbound}\n"
+            "Fire selected: {result.fire_selected}\n"
+            "Fire current: {result.fire_current}\n"
+            "WiFi Led: {result.wifi_led_setting}\n"
+            "Hardware version: {result.hardware_version}\n"
+            "Firmware version: {result.firmware_version}\n",
+        )
+    )
+    def status(self) -> IHCookerStatus:
+        """Retrieve properties."""
+        properties_new = [
+            "func",
+            "menu",
+            "action",
+            "t_func",
+            "version",
+            "profiles",
+            "set_wifi_led",
+            "play",
+        ]
+        properties_old = [
+            "func",
+            "menu",
+            "action",
+            "t_func",
+            "version",
+            "profiles",
+            "play",
+        ]
+
+        values = self.send("get_prop", ["all"])
+
+        if len(values) not in [len(properties_new), len(properties_old)]:
+            raise IHCookerException(
+                "Count (%d or %d) of requested properties does not match the "
+                "count (%s) of received values."
+                % (len(properties_new), len(properties_old), len(values)),
+            )
+        elif len(values) == len(properties_new):
+            properties = properties_new
+        elif len(values) == len(properties_old):
+            properties = properties_old
+
+        return IHCookerStatus(
+            self.model, defaultdict(lambda: None, zip(properties, values))
+        )
+
+    @command(
+        click.argument("profile", type=str),
+        click.argument("skip_confirmation", type=bool, default=False),
+        default_output=format_output("Cooking profile requested."),
+    )
+    def start(self, profile: Union[str, c.Container, dict], skip_confirmation=False):
+        """Start cooking a profile.
+
+        :arg
+
+        Please do not use skip_confirmation=True, as this is potentially unsafe.
```

Comment:
How unsafe this is? Unsafe enough to avoid exposing it? :-)

Reply:
Good question. The device itself has a couple of fail-saves: automatically stops if there's no ferrous metal on the cooktop, and overheating protection if the temp sensor measures above 250c if I'm not mistaken. I can imagine remote start can be a nice feature for certain projects. What's your stance on this?

Reply:
I think it's fair enough to keep it available, anyone toggling this on should know what they are doing.